tools/memap.py

```diff
@@ -361,56 +479,136 @@ def parse_map_file_iar(self, file_desc):
+        rex = r'^(.+)\/(.+\.map)$'
         test_rex = re.match(rex, path)
 
         if test_rex:
-            search_path = test_rex.group(1) + '/mbed-os/'
+            search_path = test_rex.group(1)
         else:
             print "Warning: this doesn't look like an mbed project"
             return
 
+        # create empty disctionary
+        self.modules = dict()
+
+        # serach for object files
         for root, _, obj_files in os.walk(search_path):
             for obj_file in obj_files:
                 if obj_file.endswith(".o"):
-                    module_name, object_name = self.path_object_to_module_name(
-                        os.path.join(root, obj_file))
-
-                    if object_name in self.object_to_module:
-                        if DEBUG:
-                            print "WARNING: multiple usages of object file: %s"\
-                                % object_name
-                            print "    Current: %s" % \
-                                self.object_to_module[object_name]
-                            print "    New:     %s" % module_name
-                            print " "
-                    else:
-                        self.object_to_module.update({object_name:module_name})
+
+                    txt = os.path.join(root, obj_file)
+
+                    txt = txt.replace('\\', '/')
+
+                    # add relative path + object to list
+                    #self.dir_obj.append(txt[len(search_path)+1:])
+                    self.module_init(txt[len(search_path)+1:])
+
+
+    def reduce_depth(self, depth):
+        """
+        prints list of directories and objects. Examples:
+
+        (1) depth = 1:
+        main.o
+        mbed-os
+
+        (2) depth = 2:
+        main.o
+        mbed-os/test.o
+        mbed-os/drivers
+
+        """
+
+        # depth 0 shows all entries
+        if depth == 0:
+            self.short_modules = deepcopy(self.modules)
+            return
+
+        self.short_modules = dict()
+
+        # create reduced list
+        for line in self.modules:
+
+            data = line.split('/')
+            ndir = len(data)
+
+            temp = ''
+            count = 0
+
+            # iterate until the max depth level
+            max_level = min(depth, ndir)
+
+            # rebuild the path based on depth level
+            while count < max_level:
+                if count > 0:    # ignore '/' from first entry
+                    temp = temp + '/'
+
+                temp = temp + data[count]
+                count += 1
+
+            if temp not in self.short_modules:
+                temp_dic = dict()
+                for section_idx in self.all_sections:
+                    temp_dic[section_idx] = 0
+                self.short_modules[temp] = temp_dic
+
+            for section_idx in self.all_sections:
+                self.short_modules[temp][section_idx] += \
+                self.modules[line][section_idx]
```

Comment:
Can you indent this line so that, when I read this file again in 2 months, I know that it's a continuation of the prior line?